### PR TITLE
added watchify and karma coverage support

### PR DESCRIPTION
**Pull request by nickalie**

### CODE REVIEW

File: Gulpfile.js [hunk 1]
```diff
@@ -318,54 +301,21 @@ gulp.task('clean', function (cb) {
...
-    var dist = [];
-    _.each(SOURCES, function (it) {
-        dist.push(it.replace('src/', 'dist/'))
-    })
-
-    return gulp.src(["dist/**/*.js", "dist/templates/*.js"])
-        .pipe(resolve())
+    return gulp.src(['dist/' + pkg.name + '.js', 'dist/templates/*.js'])
         .pipe(sourcemaps.init({loadMaps: true}))
-        .pipe(gulpif(prod, uglify({mangle: false})))
```
Comment (moshloop) [inline Gulpfile.js:339]: how has the "--env production" option been handled to only uglify in prod handled?

File: Gulpfile.js [hunk 2]
```diff
@@ -318,54 +301,21 @@ gulp.task('clean', function (cb) {
...
-        .pipe(gulpif(options.watch, pseudoconcat(main + ".js", {
-            webRoot: 'src',
-            host: 'http://localhost:' + port + '/'
-        }), concat(main + ".js")))
-        .pipe(gulpif(options.watch, replace('/dist/', '/')))
-        .pipe(gulpif(options.watch, replace('http://localhost:' + port + '/../', 'http://localhost:' + port + '/')))
-        .pipe(sourcemaps.write('.', {includeContent: !prod}))
```
Comment (moshloop) [inline Gulpfile.js:346]: sourcemaps should not include content in production builds.

Comment (nickalie) [inline Gulpfile.js:346]: @Egis-Moshe fixed

File: Gulpfile.js (hunk 1 above)
Comment (nickalie) [inline Gulpfile.js:339]: @Egis-Moshe fixed

**Pull request merged**